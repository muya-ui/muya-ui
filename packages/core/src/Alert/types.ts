import React from 'react';

import { IAnimationBaseProps } from '../Animation/types';
import { ICustomStyleBaseProps, ISizeSpecBaseProps } from '../types';

/**
 * Alert style keys
 * @styles wrapper Alert 外部容器
 * @styles container Alert 居中容器
 * @styles icon Icon 节点
 * @styles title 标题节点
 * @styles desc 描述节点
 * @styles close 关闭 Icon 节点
 */
export type IAlertStyleKeys = 'wrapper' | 'container' | 'icon' | 'title' | 'desc' | 'close';

type PropsFromAnimation = Pick<IAnimationBaseProps, 'appear' | 'timeout'>;

export interface IAlertPureProps
  extends ICustomStyleBaseProps<IAlertStyleKeys>,
    PropsFromAnimation {
  /**
   * 显示关闭按钮
   * @default false
   */
  showClose?: boolean;
  /**
   * 标题
   */
  title?: React.ReactNode;
  /**
   * 描述
   */
  description?: React.ReactNode;
  /**
   * 显示图标
   * @default true
   */
  showIcon?: boolean;
  /**
   * 类型
   * success-成功 error-失败 warning-警告 info-提示
   * @default 'info'
   */
  type?: 'success' | 'error' | 'warning' | 'info';
  /**
   * 显示与否，注意传了这个值 Alert 就变成受控组件了
   * @default true
   */
  visible?: boolean;
  /**
   * 点击关闭后的回调
   */
  onClose?: () => void;
  /**
   * children 当成是 title 使用
   */
  childrenAsTitle?: boolean;
  /**
   * 过渡组件
   * @default Animation.Collapse
   **/
  TransitionComponent?: React.ComponentType<any>;
}

export type IAlertProps = IAlertPureProps &
  ISizeSpecBaseProps &
  React.HTMLAttributes<HTMLDivElement>;
