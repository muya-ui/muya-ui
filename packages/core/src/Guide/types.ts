import React from 'react';
import { ReactourStepContentArgs } from 'reactour';

import { IPopperPlacement } from '../Popper';
import { ITriggerArrowStyle } from '../Trigger';
import { ICustomStyleBaseProps } from '../types';

/**
 * @styles content 内容
 * @styles skip 跳过
 * @styles next 下一项
 * @styles text 文本
 * @styles close 关闭
 * @styles inform inform 图标
 * @styles title 标题
 * @styles subTitle 子标题
 */
export type IGuideStyleKeys =
  | 'content'
  | 'skip'
  | 'next'
  | 'text'
  | 'close'
  | 'inform'
  | 'title'
  | 'subTitle';
export type IGuideStepPosition = 'left' | 'right' | 'top' | 'bottom';

export interface IGuideStep {
  /**
   * 引导文案或自定义引导框
   */
  content: React.ReactNode | ((args: ReactourStepContentArgs) => React.ReactNode);
  /**
   * 副标题文案
   */
  subTitle?: string;
  /**
   * DOM 选择器
   */
  selector: string;
  /**
   * 引导弹出位置。如果选择的 DOM 靠近屏幕右侧边界且 position 为 right，引导框位置将会自适应改变为 left
   * @default bottom
   */
  position?: IGuideStepPosition;
  /**
   * 箭头指示位置。特殊情况下需要自定义箭头位置，比如选择的 DOM 靠近屏幕右侧边界
   * @default `${position}-start`
   */
  arrowPosition?: IPopperPlacement;
}

export interface IGuideProps extends ICustomStyleBaseProps<IGuideStyleKeys> {
  /**
   * 引导步骤配置
   */
  steps: Array<IGuideStep>;
  /**
   * 关闭事件
   */
  onRequestClose: () => void;
  /**
   * 自定义箭头大小、颜色
   */
  arrowStyle?: ITriggerArrowStyle;
  /**
   * 是否显示 Close Icon
   * @default true
   */
  showClose?: boolean;
  /**
   * 是否显示跳过文案
   * @default true
   */
  showSkip?: boolean;
  /**
   * 跳过文案
   * @default 跳过引导
   */
  skipText?: string;
  /**
   * 下一步文案
   * @default 下一步(index/total)
   */
  nextText?: string;
  /**
   * 确认完成引导文案
   * @default 完成
   */
  confirmText?: string;
  /**
   * 高亮框边距
   * @default 0
   */
  maskSpace?: number;
  /**
   * 高亮框圆角
   * @default 0
   */
  rounded?: number;
  /**
   * 点击蒙层是否触发关闭事件
   * @default false
   */
  closeWithMask?: boolean;
  /**
   * 蒙层的 className
   */
  maskClassName?: string;
  /**
   * tooltips的 className
   */
  tooltipsClassName?: string;
  /**
   * selector 元素是否可以进行交互
   * @default true
   */
  disableInteraction?: boolean;
}
