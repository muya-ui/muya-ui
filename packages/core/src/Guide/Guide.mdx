---
name: Guide 新功能引导
menu: 组件
route: /components/guide
autoDemos: true
parent: 业务组件
---

# Guide 新功能引导

新功能上线或新内容更新，需要一定程度的引导，引起用户关注，起到告知或强调的作用。

## 使用场景

根据功能类型不同，希望强调的程度不同，需采用不同程度的新功能引导，切莫对用户造成过度的干扰。

{Demos}

### API

#### IGuideProps

<Props propsName="IGuideProps" />

#### IGuideStep

<Props propsName="IGuideStep" />

#### IGuideStepPosition

```typescript
type IGuideStepPosition = 'left' | 'right' | 'top' | 'bottom';
```

### Styles

#### IGuideStyleKeys

<Styles typeName="IGuideStyleKeys" componentName="guide" />
