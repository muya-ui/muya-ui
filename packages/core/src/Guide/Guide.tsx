import React, { forwardRef, useCallback, useState } from 'react';
import Tour from 'reactour';

import { useLockScroll } from '@muya-ui/utils';
import { CloseIcon, InformIcon } from '@muya-ui/theme-light';

import { Button } from '../Button';
import useLocale from '../Locale/useLocale';
import { IPopperPlacement } from '../Popper';
import { StyledArrow } from '../styled/components/Arrow';
import { Typography } from '../Typography';
import useStyles from '../utils/useStyles';
import useTheme from '../utils/useTheme';
import {
  ButtonWrapper,
  CloseIconWrapper,
  GlobalStyle,
  SkipWrapper,
  StyledIcon,
  TextWrapper,
  TitleWrapper,
  TooltipsArrowStyle,
  ToolTipsContent,
  TourWrapper,
} from './styled';
import { IGuideProps } from './types';

const defaultStyles = {
  content: '',
  text: '',
  next: '',
  close: '',
  skip: '',
  inform: '',
  title: '',
  subTitle: '',
};

const Guide = forwardRef<Tour, IGuideProps>((props, ref) => {
  const local = useLocale();
  const theme = useTheme();
  const {
    onRequestClose,
    arrowStyle,
    steps,
    showClose = true,
    showSkip = true,
    skipText = local['Guide.skip'],
    confirmText = local['Guide.confirm'],
    nextText,
    styles,
    maskSpace = 0,
    rounded = 0,
    closeWithMask = false,
    maskClassName = `${theme.prefix}-guide-mask`,
    tooltipsClassName = `${theme.prefix}-guide-tooltips`,
    disableInteraction = true,
  } = props;
  const innerStyles = useStyles('guide', defaultStyles, styles);
  const [step, setStep] = useState(0);
  const [lock, setLock] = useState(false);
  useLockScroll(lock);

  const goToNext = useCallback(
    (index: number, close: () => void) => {
      if (index + 1 === steps.length) {
        close();
      } else {
        setStep(index + 1);
      }
    },
    [steps.length],
  );

  const Inform = theme.components.Guide.informIcon.Icon || InformIcon;
  const Close = theme.components.Guide.closeIcon.Icon || CloseIcon;

  steps.map((step, index) => {
    const buttonTitle =
      index === steps.length - 1
        ? confirmText
        : nextText || `${local['Guide.nextStep']}(${index + 1}/${steps.length})`;
    if (!step.position) step.position = 'bottom';
    if (typeof step.content === 'string') {
      let placement: IPopperPlacement = 'bottom-start';
      switch (step.position) {
        case 'top': {
          placement = 'top-start';
          break;
        }
        case 'left': {
          placement = 'left-start';
          break;
        }
        case 'right': {
          placement = 'right-start';
          break;
        }
      }
      let content = step.content;
      step.content = ({ close }) => (
        <ToolTipsContent theme={theme} {...innerStyles.content}>
          <TooltipsArrowStyle
            $placement={step.position}
            $tooltipsClassName={tooltipsClassName}
            theme={theme}
          />
          <TextWrapper {...innerStyles.text}>
            <StyledIcon theme={theme} {...innerStyles.inform}>
              <Inform />
            </StyledIcon>
            <TitleWrapper>
              <Typography.Title level={theme.components.Guide.titleLevel} {...innerStyles.title}>
                {content}
              </Typography.Title>
              {step.subTitle ? (
                <Typography.Text color="assistant" {...innerStyles.subTitle}>
                  {step.subTitle}
                </Typography.Text>
              ) : null}
            </TitleWrapper>
          </TextWrapper>
          <ButtonWrapper theme={theme} {...innerStyles.next}>
            <Button onClick={_ => goToNext(index, close)} type="primary">
              {buttonTitle}
            </Button>
          </ButtonWrapper>
          <StyledArrow placement={step.arrowPosition || placement} arrowStyle={arrowStyle} />
          {showClose ? (
            <CloseIconWrapper
              weakLevel={1}
              type="weak"
              theme={theme}
              onClick={close}
              {...innerStyles.close}
            >
              <Close fontSize={theme.components.Guide.closeIcon.fontSize} />
            </CloseIconWrapper>
          ) : null}
          {showSkip ? (
            <SkipWrapper theme={theme} onClick={close} {...innerStyles.skip}>
              <Typography.Text color="disabled">{skipText}</Typography.Text>
            </SkipWrapper>
          ) : null}
        </ToolTipsContent>
      );
    }
    return step;
  });

  const disableBody = () => setLock(true);
  const enableBody = () => setLock(false);

  return (
    <>
      <TourWrapper
        onRequestClose={onRequestClose}
        theme={theme}
        steps={steps}
        isOpen
        maskClassName={maskClassName}
        className={tooltipsClassName}
        showNavigation={false}
        showButtons={false}
        showCloseButton={false}
        showNumber={false}
        goToStep={step}
        maskSpace={maskSpace}
        rounded={rounded}
        onAfterOpen={disableBody}
        onBeforeClose={enableBody}
        closeWithMask={closeWithMask}
        disableInteraction={disableInteraction}
        ref={ref}
      />
      <GlobalStyle theme={theme} />
    </>
  );
});

export default Guide;
