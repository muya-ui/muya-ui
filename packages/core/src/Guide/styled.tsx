import React from 'react';
import Tour from 'reactour';
import styled, { createGlobalStyle, css } from 'styled-components';

import { InlineButton } from '../Button';
import { IThemedBaseProps } from '../types';
import { StyledTitle } from '../Typography/Title';
import { IGuideStepPosition } from './types';

interface IStyledProps extends IThemedBaseProps {
  $placement?: IGuideStepPosition;
  $tooltipsClassName?: string;
}

export const CloseIconWrapper = styled(InlineButton)`
  ${(props: IThemedBaseProps) => {
    const { top, right, color } = props.theme.components.Guide.closeIcon;
    return css`
      position: absolute;
      right: ${right}px;
      top: ${top}px;
      color: ${color};
    `;
  }}
`;

export const ToolTipsContent = styled.div`
  ${(props: IThemedBaseProps) => {
    const { padding, minWidth } = props.theme.components.Guide.toolTips;
    return css`
      position: relative;
      padding: ${padding};
      min-width: ${minWidth}px;
      ${StyledTitle} {
        margin-bottom: 0;
      }
    `;
  }}
`;

export const ButtonWrapper = styled.div`
  ${(props: IThemedBaseProps) => {
    return css`
      margin-top: ${props.theme.components.Guide.nextButton.marginTop}px;
      display: flex;
      flex-direction: row-reverse;
    `;
  }}
`;

export const GlobalStyle = createGlobalStyle`
  /*
    这么写是为了防止 styled-components 产生相同的ID，
    详见：https://github.com/styled-components/styled-components/issues/3097
  */
  #___reactour {
    & > div:first-child {
      opacity: ${props => props.theme.components.Guide.mask.opacity};
      color: ${props => props.theme.components.Guide.mask.color};
    }
  }
`;

function getSpaceByPlacementOnTop(props: IStyledProps) {
  const { space } = props.theme.components.Guide.toolTips;
  if (props.$placement === 'top') {
    return -space;
  }
  return space;
}
/*
  这么写是为了防止 styled-components 产生相同的ID，
  详见：https://github.com/styled-components/styled-components/issues/3097
*/
const ToolTipsArrowOnTopStyle = createGlobalStyle`
  /* 箭头在上的样式 */
  .${(props: IStyledProps) => props.$tooltipsClassName} {
    top: ${getSpaceByPlacementOnTop}px !important;
  }
`;
function getSpaceByPlacementOnLeft(props: IStyledProps) {
  const { space } = props.theme.components.Guide.toolTips;
  if (props.$placement === 'left') {
    return -space;
  }
  return space;
}
const ToolTipsArrowOnLeftStyle = createGlobalStyle`
  /* 箭头在左的样式 */
  .${(props: IStyledProps) => props.$tooltipsClassName} {
    left: ${getSpaceByPlacementOnLeft}px !important;
  }
`;

export const TooltipsArrowStyle = (props: IStyledProps) => {
  if (props.$placement === 'left' || props.$placement === 'right') {
    return <ToolTipsArrowOnLeftStyle {...props} />;
  }
  return <ToolTipsArrowOnTopStyle {...props} />;
}

export const TourWrapper = styled(Tour)`
  ${(props: IThemedBaseProps) => {
    const { borderRadius, boxShadow } = props.theme.components.Guide.toolTips;
    return css`
      padding: 0;
      border-radius: ${borderRadius}px;
      box-shadow: ${boxShadow};
    `;
  }}
`;

export const SkipWrapper = styled.div`
  ${(props: IThemedBaseProps) => {
    return css`
      position: absolute;
      left: 0;
      right: 0;
      bottom: ${props.theme.components.Guide.skip.bottom}px;
      margin: 0 auto;
      text-align: center;

      &:hover {
        cursor: pointer;
      }
    `;
  }}
`;

export const TextWrapper = styled.div`
  display: flex;
  align-items: baseline;
`;

export const StyledIcon = styled.span`
  ${(props: IThemedBaseProps) => {
    const { marginRight, color } = props.theme.components.Guide.informIcon;
    return css`
      margin-right: ${marginRight}px;
      color: ${color};
    `;
  }}
`;

export const TitleWrapper = styled.div``;
