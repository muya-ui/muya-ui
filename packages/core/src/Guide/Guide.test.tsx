import React from 'react';
import Guide from './Guide';
import { mount } from 'enzyme';
import { IGuideStep } from './types';

describe('Guide', () => {
  it('should render success', function() {
    const tourConfig = (position = 'right') => [
      // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
      {
        selector: '.step1',
        position,
        content: '这是第一个新功能',
        subTitle: '这是副标题',
        arrowPosition: 'right-end',
      } as IGuideStep,
    ];
    const wrapper = mount(
      <>
        <Guide steps={tourConfig('left')} onRequestClose={() => {}} />
      </>,
    );
    const wrapper1 = mount(
      <>
        <Guide
          steps={tourConfig('top')}
          confirmText="确定啦"
          nextText="下一步"
          skipText="跳过啦"
          onRequestClose={() => {}}
        />
      </>,
    );
    const wrapper2 = mount(
      <>
        <Guide steps={tourConfig()} showSkip={false} showClose={false} onRequestClose={() => {}} />
      </>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper1.html()).toMatchSnapshot();
    expect(wrapper2.html()).toMatchSnapshot();
  });
});
