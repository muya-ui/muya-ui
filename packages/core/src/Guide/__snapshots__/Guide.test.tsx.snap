// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guide should render success 1`] = `"<div class=\\"sc-dnqmqq cVioWK\\"><svg width=\\"0\\" height=\\"0\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"muya-guide-mask\\"><defs><mask id=\\"mask-main\\"><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"black\\"></rect><rect x=\\"-1\\" y=\\"-1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"1\\" y=\\"-1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"-1\\" y=\\"1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"1\\" y=\\"1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black \\"></circle></mask><clipPath id=\\"clip-path\\"><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect></clipPath></defs><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"currentColor\\" mask=\\"url(#mask-main)\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"currentColor\\" clip-path=\\"url(#clip-path)\\" pointer-events=\\"auto\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" pointer-events=\\"auto\\" fill=\\"transparent\\" display=\\"block\\" class=\\"reactour__mask--disable-interaction undefined\\"></rect></svg></div><div data-focus-guard=\\"true\\" tabindex=\\"0\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div><div data-focus-guard=\\"true\\" tabindex=\\"1\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div><div data-focus-lock-disabled=\\"false\\"><div tabindex=\\"-1\\" class=\\"sc-ifAKCX dUjRvg reactour__helper TourWrapper-jIgpNd fKcYpK muya-guide-tooltips reactour__helper--is-open\\" role=\\"dialog\\"><div class=\\"ToolTipsContent-dnCJNT cKDyHA muya-guide-content\\"><div class=\\"TextWrapper-FmaIz eQrRnS muya-guide-text\\"><span class=\\"StyledIcon-kFlXVF WfonH muya-guide-inform\\"><svg focusable=\\"false\\" aria-hidden=\\"false\\" viewBox=\\"0 0 1024 1024\\" class=\\"StyledSvg-gOpsDq cgzRxQ\\"><path d=\\"M853.333 704h64a21.333 21.333 0 0121.334 21.333V832a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 832V725.333A21.333 21.333 0 01106.667 704h64V341.333C170.667 152.821 323.488 0 512 0c188.512 0 341.333 152.821 341.333 341.333V704zm-468.81 213.333h254.944C633.6 977.131 578.773 1024 512 1024s-121.6-46.87-127.477-106.667z\\"></path></svg></span><div class=\\"TitleWrapper-cNIjFy illxcA\\"><h5 class=\\"StyledTitle-ewyitM fkuXKS muya-guide-title\\">这是第一个新功能</h5><span color=\\"assistant\\" class=\\"Text-bbIbrg eRxpWA muya-guide-sub-title\\">这是副标题</span></div></div><div class=\\"ButtonWrapper-iIRbUd htbuEB muya-guide-next\\"><button type=\\"button\\" class=\\"StyledButton-dSfbpv cagvoq Button-hYXUXp dlqAdK\\">完成</button></div><div class=\\"StyledArrow-dxEYPw hHMKzc\\"></div><button type=\\"button\\" class=\\"StyledButton-dSfbpv cagvoq InlineButtonWithoutTheme-hGlcsL CloseIconWrapper-kYqSEh ieXoCd muya-guide-close\\"><svg focusable=\\"false\\" aria-hidden=\\"false\\" viewBox=\\"0 0 1024 1024\\" class=\\"StyledSvg-gOpsDq dgxAAa\\" font-size=\\"10\\"><path d=\\"M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z\\"></path></svg></button><div class=\\"SkipWrapper-fOLqyA jzNYXq muya-guide-skip\\"><span color=\\"disabled\\" class=\\"Text-bbIbrg hEKcOP\\">跳过引导</span></div></div></div></div><div data-focus-guard=\\"true\\" tabindex=\\"0\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div>"`;

exports[`Guide should render success 2`] = `"<div class=\\"sc-dnqmqq cVioWK\\"><svg width=\\"0\\" height=\\"0\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"muya-guide-mask\\"><defs><mask id=\\"mask-main\\"><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"black\\"></rect><rect x=\\"-1\\" y=\\"-1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"1\\" y=\\"-1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"-1\\" y=\\"1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"1\\" y=\\"1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black \\"></circle></mask><clipPath id=\\"clip-path\\"><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect></clipPath></defs><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"currentColor\\" mask=\\"url(#mask-main)\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"currentColor\\" clip-path=\\"url(#clip-path)\\" pointer-events=\\"auto\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" pointer-events=\\"auto\\" fill=\\"transparent\\" display=\\"block\\" class=\\"reactour__mask--disable-interaction undefined\\"></rect></svg></div><div data-focus-guard=\\"true\\" tabindex=\\"0\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div><div data-focus-guard=\\"true\\" tabindex=\\"1\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div><div data-focus-lock-disabled=\\"false\\"><div tabindex=\\"-1\\" class=\\"sc-ifAKCX dUjRvg reactour__helper TourWrapper-jIgpNd fKcYpK muya-guide-tooltips reactour__helper--is-open\\" role=\\"dialog\\"><div class=\\"ToolTipsContent-dnCJNT cKDyHA muya-guide-content\\"><div class=\\"TextWrapper-FmaIz eQrRnS muya-guide-text\\"><span class=\\"StyledIcon-kFlXVF WfonH muya-guide-inform\\"><svg focusable=\\"false\\" aria-hidden=\\"false\\" viewBox=\\"0 0 1024 1024\\" class=\\"StyledSvg-gOpsDq cgzRxQ\\"><path d=\\"M853.333 704h64a21.333 21.333 0 0121.334 21.333V832a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 832V725.333A21.333 21.333 0 01106.667 704h64V341.333C170.667 152.821 323.488 0 512 0c188.512 0 341.333 152.821 341.333 341.333V704zm-468.81 213.333h254.944C633.6 977.131 578.773 1024 512 1024s-121.6-46.87-127.477-106.667z\\"></path></svg></span><div class=\\"TitleWrapper-cNIjFy illxcA\\"><h5 class=\\"StyledTitle-ewyitM fkuXKS muya-guide-title\\">这是第一个新功能</h5><span color=\\"assistant\\" class=\\"Text-bbIbrg eRxpWA muya-guide-sub-title\\">这是副标题</span></div></div><div class=\\"ButtonWrapper-iIRbUd htbuEB muya-guide-next\\"><button type=\\"button\\" class=\\"StyledButton-dSfbpv cagvoq Button-hYXUXp dlqAdK\\">确定啦</button></div><div class=\\"StyledArrow-dxEYPw hHMKzc\\"></div><button type=\\"button\\" class=\\"StyledButton-dSfbpv cagvoq InlineButtonWithoutTheme-hGlcsL CloseIconWrapper-kYqSEh ieXoCd muya-guide-close\\"><svg focusable=\\"false\\" aria-hidden=\\"false\\" viewBox=\\"0 0 1024 1024\\" class=\\"StyledSvg-gOpsDq dgxAAa\\" font-size=\\"10\\"><path d=\\"M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z\\"></path></svg></button><div class=\\"SkipWrapper-fOLqyA jzNYXq muya-guide-skip\\"><span color=\\"disabled\\" class=\\"Text-bbIbrg hEKcOP\\">跳过啦</span></div></div></div></div><div data-focus-guard=\\"true\\" tabindex=\\"0\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div>"`;

exports[`Guide should render success 3`] = `"<div class=\\"sc-dnqmqq cVioWK\\"><svg width=\\"0\\" height=\\"0\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"muya-guide-mask\\"><defs><mask id=\\"mask-main\\"><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"black\\"></rect><rect x=\\"-1\\" y=\\"-1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"1\\" y=\\"-1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"-1\\" y=\\"1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black\\"></circle><rect x=\\"1\\" y=\\"1\\" width=\\"0\\" height=\\"0\\" fill=\\"white\\"></rect><circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"black \\"></circle></mask><clipPath id=\\"clip-path\\"><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\"></rect></clipPath></defs><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"currentColor\\" mask=\\"url(#mask-main)\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" fill=\\"currentColor\\" clip-path=\\"url(#clip-path)\\" pointer-events=\\"auto\\"></rect><rect x=\\"0\\" y=\\"0\\" width=\\"0\\" height=\\"0\\" pointer-events=\\"auto\\" fill=\\"transparent\\" display=\\"block\\" class=\\"reactour__mask--disable-interaction undefined\\"></rect></svg></div><div data-focus-guard=\\"true\\" tabindex=\\"0\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div><div data-focus-guard=\\"true\\" tabindex=\\"1\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div><div data-focus-lock-disabled=\\"false\\"><div tabindex=\\"-1\\" class=\\"sc-ifAKCX dUjRvg reactour__helper TourWrapper-jIgpNd fKcYpK muya-guide-tooltips reactour__helper--is-open\\" role=\\"dialog\\"><div class=\\"ToolTipsContent-dnCJNT cKDyHA muya-guide-content\\"><div class=\\"TextWrapper-FmaIz eQrRnS muya-guide-text\\"><span class=\\"StyledIcon-kFlXVF WfonH muya-guide-inform\\"><svg focusable=\\"false\\" aria-hidden=\\"false\\" viewBox=\\"0 0 1024 1024\\" class=\\"StyledSvg-gOpsDq cgzRxQ\\"><path d=\\"M853.333 704h64a21.333 21.333 0 0121.334 21.333V832a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 832V725.333A21.333 21.333 0 01106.667 704h64V341.333C170.667 152.821 323.488 0 512 0c188.512 0 341.333 152.821 341.333 341.333V704zm-468.81 213.333h254.944C633.6 977.131 578.773 1024 512 1024s-121.6-46.87-127.477-106.667z\\"></path></svg></span><div class=\\"TitleWrapper-cNIjFy illxcA\\"><h5 class=\\"StyledTitle-ewyitM fkuXKS muya-guide-title\\">这是第一个新功能</h5><span color=\\"assistant\\" class=\\"Text-bbIbrg eRxpWA muya-guide-sub-title\\">这是副标题</span></div></div><div class=\\"ButtonWrapper-iIRbUd htbuEB muya-guide-next\\"><button type=\\"button\\" class=\\"StyledButton-dSfbpv cagvoq Button-hYXUXp dlqAdK\\">完成</button></div><div class=\\"StyledArrow-dxEYPw hHMKzc\\"></div></div></div></div><div data-focus-guard=\\"true\\" tabindex=\\"0\\" style=\\"width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;\\"></div>"`;
